#!/usr/bin/env python
PACKAGE = "ltu_actor_route_sign_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("max_size",            int_t,      0, "maximum size in pixels considered a sign",                   160,    30,   300)
gen.add("min_size",            int_t,      0, "minimum size in pixels considered a sign",                   20,    5,    200)
gen.add("scale_factor",        double_t,   0, "higher: faster, lower: more accurate",                       1.1187, 1.01, 2.0)
gen.add("min_neighbors",       int_t,      0, "defines quality, higher = less detections, better quality",  9,      2,    20)
gen.add("trigger_width",    int_t,      0, "when the sign reaches this width, send trigger",           28,  5,    200)

gen.add("red_mask",    bool_t,      0, "turn on/off redmask", False)
gen.add("hue_lower",    int_t,      0, "hue lower bound",           28,  0,    255)
gen.add("saturation_lower",    int_t,      0, "saturation lower bound",           28,  0,    255)
gen.add("value_lower",    int_t,      0, "value lower bound",           28,  0,    255)
gen.add("hue_upper",    int_t,      0, "hue upper bound",           28,  0,    255)
gen.add("saturation_upper",    int_t,      0, "saturation upper bound",           28,  0,    255)
gen.add("value_upper",    int_t,      0, "value upper bound",           28,  0,    255)

exit(gen.generate(PACKAGE, "ltu_actor_route_sign_detection", "StopSign"))
